cmake/options.cmake

```diff
@@ -63,16 +63,25 @@ if (NOT ${MACHINE} STREQUAL "zynqmp_r5")
  set (WITH_LOAD_FW OFF)
 endif(NOT ${MACHINE} STREQUAL "zynqmp_r5")
 
-option (WITH_VIRTIO_MASTER "Build with virtio master enabled" ON)
-option (WITH_VIRTIO_SLAVE "Build with virtio slave enabled" ON)
-
-if (NOT WITH_VIRTIO_MASTER)
-  add_definitions(-DVIRTIO_SLAVE_ONLY)
-endif (NOT WITH_VIRTIO_MASTER)
-
-if (NOT WITH_VIRTIO_SLAVE)
-  add_definitions(-DVIRTIO_MASTER_ONLY)
-endif (NOT WITH_VIRTIO_SLAVE)
+option (WITH_VIRTIO_DRIVER "Build with virtio driver (front end)  enabled" ON)
+option (WITH_VIRTIO_DEVICE "Build with virtio device (back end)  enabled" ON)
+option (WITH_VIRTIO_MASTER "Build with virtio driver (front end)  enabled" ON)
+option (WITH_VIRTIO_SLAVE "Build with virtio device (back end)  enabled" ON)
+
+if (WITH_VIRTIO_MASTER)
+  message(DEPRECATION "deprecated cmake option replaced by WITH_VIRTIO_DRIVER" ...)
+endif (WITH_VIRTIO_MASTER)
+if (WITH_VIRTIO_SLAVE)
+  message(DEPRECATION "deprecated cmake option replaced by WITH_VIRTIO_DEVICE" ...)
+endif (WITH_VIRTIO_SLAVE)
+
+if (NOT WITH_VIRTIO_DRIVER AND NOT WITH_VIRTIO_MASTER)
+	add_definitions(-DVIRTIO_DEVICE_ONLY)
+endif (NOT WITH_VIRTIO_DRIVER AND NOT WITH_VIRTIO_MASTER)
+
+if (NOT WITH_VIRTIO_DEVICE AND NOT WITH_VIRTIO_SLAVE)
+	add_definitions(-DVIRTIO_DRIVER_ONLY)
+endif (NOT WITH_VIRTIO_DEVICE AND NOT WITH_VIRTIO_SLAVE)
```

Comment:
> Building generate dprecation warnings by default as `WITH_VIRTIO_MASTER` and `WITH_VIRTIO_SLAVE` are enabled by default. following code seems working better:
> 
> ```
> if (NOT WITH_VIRTIO_MASTER)
>   message(DEPRECATION "deprecated WITH_VIRTIO_MASTER option replaced by WITH_VIRTIO_DRIVER" ...)
> endif (NOT WITH_VIRTIO_MASTER)
> if (NOT WITH_VIRTIO_SLAVE)
>   message(DEPRECATION "deprecated WITH_VIRTIO_SLAVE option replaced by WITH_VIRTIO_DEVICE" ...)
> endif (NOT WITH_VIRTIO_SLAVE)
> 
> if (NOT WITH_VIRTIO_DRIVER OR NOT WITH_VIRTIO_MASTER)
>  	add_definitions(-DVIRTIO_DEVICE_ONLY)
> endif (NOT WITH_VIRTIO_DRIVER OR NOT WITH_VIRTIO_MASTER)
>  
> if (NOT WITH_VIRTIO_DEVICE OR NOT WITH_VIRTIO_SLAVE)
>  	add_definitions(-DVIRTIO_DRIVER_ONLY)
> endif (NOT WITH_VIRTIO_DEVICE OR NOT WITH_VIRTIO_SLAVE)
> ```
This comment as not taken into account

Reply:
I just defaulted the deprecated variables to "OFF" instead of "ON". That gets rid of the warnings on a normal build.
It seems like the top half just warns if the offending variables are turned off, which seems backwards to me.

Reply:
You are right, I have re-tested in different configurations (mainly on zephyr), it works. 
My test shows a potential side effect: VIRTIO_DEVICE_ONLY or DVIRTIO_DRIVER_ONLY may not be configured 
an increase of the library.